server/app/routes/user/user.js

```diff
@@ -3,22 +3,37 @@ const router = require('express').Router();
 const db = require('../../../db');
 const User = db.model('user');
 const Promise = require('bluebird');
+const HttpError = require('../../../utils/HttpError');
+var chalk = require('chalk');
 
+router.param('userId', function (req, res, next, id) {
+
+  User.findById(id)
+  .then(function (user) {
+    if (!user) throw HttpError(404);
+    req.requestedUser = user;
+    next();
+  })
+  .catch(next);
+});
+
+function assertIsCorrectUser (req, res, next) {
+  if (+req.user.id === +req.requestedUser.id) next();
+  else next(HttpError(401));
+}
+
+router.use(['/:userId/\*','/:userId'],assertIsCorrectUser);
 
```

Comment:
Do we need this array?  It won't hit in on every /:userId route without the wildcard?